components/ledger/ledger-utxo-flow/src/main/kotlin/net/corda/ledger/utxo/flow/impl/flows/backchain/TransactionBackchainResolutionFlow.kt

```diff
@@ -54,17 +55,17 @@ class TransactionBackchainResolutionFlowVersionedFlowFactory(
 
     override fun create(version: Int, session: FlowSession): SubFlow<Unit> {
         return when {
-            version >= 50100 -> TransactionBackchainResolutionFlowV1(
+            version >= CORDA_5_1.value -> TransactionBackchainResolutionFlowV1(
                 initialTransactionIds,
                 session,
                 TransactionBackChainResolutionVersion.V2
             )
-            version in 1..50099 -> TransactionBackchainResolutionFlowV1(
+            version in 1 until CORDA_5_1.value -> TransactionBackchainResolutionFlowV1(
```

Comment:
```suggestion
            version in CORDA_5_0.value until CORDA_5_1.value -> TransactionBackchainResolutionFlowV1(
```

Would this make more sense logically?

Reply:
If we want to cover the `1 until CORDA_5_0.value` range somehow, I think the V1 flow is friendlier than the exception.
On the other hand, it should not make any difference for real/production systems.

Reply:
Yeah, I couldn't see a scenario where it would make sense that we would cover 1 up to 5.0 - effectively the enum definitions are replacing the integer values. So aren't we interested in the first version (ordinal 0 in the enum)?

It's not a big deal, it just looks a bit more readable and consistent to me.

Reply:
The usage of enums is not enforced. The platform version will come from "somewhere else".
In the current form, the PR prepares the code for as many integers as possible and tries to handle every case in the best possible way. But I guess, this is a bit subjective.

Reply:
I guess it could be enforced by changing version from an Int to the enum type, but unfortunately that would be a breaking change. Or a non-breaking way to do it would be to throw an exception if the supplied version is not in our enumeration.

If we can't enforce the version list, then it compromises the ability for the version to be a "source of truth" - all code must assume it is dealing with any arbitrary version. The enumeration is simply a convenience helper to identify "well known" Corda versions. If we're happy with that, then I guess that's ok.

Reply:
I have aimed for this second target. (convenience helper)
It is not perfect. Just a step forward.